### PR TITLE
Webpack VSCode extension

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Description
- Webpack the extension, and minify with Terser
   - Size from 26MB->5MB
   - Startup time from over 2 seconds on my laptop to barley perceivable
   - Number of files 4400 -> 75
- output directory from `out` to `dist`
- Dev experience is impaired due to lack of watch

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/shared/defaultAwsClientBuilder.test.ts [hunk 1]
```diff
@@ -17,29 +17,12 @@ describe('DefaultAwsClientBuilder', () => {
             }
         }
 
-        // We don't want to test against the versions reported by package.json and vscode.version--this is
-        // what the product code does, so these tests would still pass with bad versions. Instead, we
-        // verify that a valid semver is used. This protects us against things like `null`, `undefined`,
-        // or other unexpected values.
-        const semverRegex = (require('semver-regex') as () => RegExp)()
```
Comment (awschristou) [inline src/test/shared/defaultAwsClientBuilder.test.ts]: Can `semver-regex` be removed from package.json?

File: package.json [hunk 2]
```diff
@@ -443,19 +443,20 @@
...
         "bundleDeps": "node ./build-scripts/bundleDeps.js",
-        "clean": "node ./build-scripts/clean.js out",
+        "clean": "node ./build-scripts/clean.js dist",
         "generateNonCodeFiles": "ts-node ./build-scripts/generateNonCodeFiles.ts",
-        "reset": "node ./build-scripts/clean.js out node_modules && npm install",
+        "reset": "node ./build-scripts/clean.js dist node_modules && npm install",
         "copyNonCodeFiles": "node ./build-scripts/copyNonCodeFiles.js",
         "buildScripts": "npm run bundleDeps && npm run copyNonCodeFiles && npm run generateNonCodeFiles",
-        "compile": "tsc -p ./ && npm run lint && npm run buildScripts",
+        "compile": " webpack --mode development && npm run buildScripts",
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply package.json]: Fixed

File: src/shared/constants.ts [hunk 3]
```diff
@@ -3,8 +3,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { NpmPackage } from './npmPackage'
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply src/shared/constants.ts]: Yep, deleted

File: src/test/shared/defaultAwsClientBuilder.test.ts (hunk 1 above)
Comment (hunterwerlla) [reply src/test/shared/defaultAwsClientBuilder.test.ts]: Yep it can, also re-added the test with updated stuff

File: package-lock.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "aws-toolkit-vscode",
-    "version": "1.4.0",
+    "version": "1.5.0",
```
Comment (awschristou) [inline package-lock.json]: This should be 1.4.0

File: package.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "name": "aws-toolkit-vscode",
     "displayName": "AWS Toolkit for Visual Studio Code",
     "description": "An extension for working with Amazon Web Services in Visual Studio Code",
-    "version": "1.4.0",
+    "version": "1.5.0",
```
Comment (awschristou) [inline package.json]: 1.4.0?

File: package-lock.json (hunk 4 above)
Comment (hunterwerlla) [reply package-lock.json]: hm that's weird, fixed in both places

**Pull request merged**